test/test_functional.py

```diff
@@ -0,0 +1,182 @@
+            [[4., 0.], [4., 0.], [4., 0.], [4., 0.], [4., 0.]],
+            [[0., 0.], [0., 0.], [0., 0.], [0., 0.], [0., 0.]],
+            [[0., 0.], [0., 0.], [0., 0.], [0., 0.], [0., 0.]]
+        ])
+
+        estimate = torchaudio.functional.istft(stft, n_fft=4, length=4)
+        self._compare_estimate(torch.ones(4), estimate)
+
+    def test_istft_of_zeros(self):
+        # stft = torch.stft(torch.zeros(4), 4)
+        stft = torch.zeros((3, 5, 2))
+
+        estimate = torchaudio.functional.istft(stft, n_fft=4, length=4)
+        self._compare_estimate(torch.zeros(4), estimate)
+
+    def test_istft_requires_overlap_windows(self):
+        # the window is size 1 but it hops 20 so there is a gap which throw an error
+        stft = torch.rand((3, 5, 2))
+        self.assertRaises(AssertionError, torchaudio.functional.istft, stft, n_fft=4,
+                          hop_length=20, win_length=1, window=torch.ones(1))
+
+    def test_istft_requires_nola(self):
+        stft = torch.zeros((3, 5, 2))
+        kwargs_ok = {
+            'n_fft': 4,
+            'win_length': 4,
+            'window': torch.ones(4),
+        }
+
+        kwargs_not_ok = {
+            'n_fft': 4,
+            'win_length': 4,
+            'window': torch.zeros(4),
+        }
+
+        # A window of ones meets NOLA but a window of zeros does not. This should
+        # throw an error.
+        torchaudio.functional.istft(stft, **kwargs_ok)
+        self.assertRaises(AssertionError, torchaudio.functional.istft, stft, **kwargs_not_ok)
+
+    def _test_istft_of_sine(self, amplitude, L, n):
+        # stft of amplitude*sin(2*pi/L*n*x) with the hop length and window size equaling L
+        x = torch.arange(2 * L, dtype=torch.get_default_dtype())
+        sound = amplitude * torch.sin(2 * math.pi / L * x * n)
+        # stft = torch.stft(sound, L, hop_length=L, win_length=L,
+        #                   window=torch.ones(L), center=False, normalized=False)
+        stft = torch.zeros((L // 2 + 1, 2, 2))
+        stft_largest_val = (amplitude * L) / 2.0
+        if n < stft.size(0):
+            stft[n, :, 1] = -stft_largest_val
+
+        if 0 <= L - n < stft.size(0):
+            # symmetric about L // 2
+            stft[L - n, :, 1] = stft_largest_val
+
+        estimate = torchaudio.functional.istft(stft, L, hop_length=L, win_length=L,
+                                               window=torch.ones(L), center=False, normalized=False)
+        self._compare_estimate(sound, estimate, atol=1e-3)
+
+    def test_istft_of_sine(self):
+        self._test_istft_of_sine(amplitude=123, L=5, n=1)
+        self._test_istft_of_sine(amplitude=150, L=5, n=2)
+        self._test_istft_of_sine(amplitude=111, L=5, n=3)
+        self._test_istft_of_sine(amplitude=160, L=7, n=4)
+        self._test_istft_of_sine(amplitude=145, L=8, n=5)
+        self._test_istft_of_sine(amplitude=80, L=9, n=6)
+        self._test_istft_of_sine(amplitude=99, L=10, n=7)
```

Comment:
The frequency n is becoming high relative to the length: we should expect to have at least 5 points per oscillation to resolve a wave properly, depending on how the inverse fourier transform works. This might be a source of higher error.

Reply:
No increasing the number of points does not reduce absolute error. Decreasing the amplitude to 1 does though (absolute error of 1e-3 to 1e-5)